File: remove-duplicates-from-sorted-array-ii.py
class Solution:
    # @param A a list of integers
    # @return an integer
    def removeDuplicates(self, A):
        A.sort()
        l = len(A)
        cnt = 0
        for i in range(0, l):
            if i == 0:
                pre2 = A[i]
                cnt = cnt + 1
            elif i == 1:
                pre1 = A[i]
                cnt = cnt + 1
            else:
                if pre1 == A[i] and pre2 == A[i]:
                    pre2 = pre1
                    pre1 = A[i]
                    continue
                pre2 = pre1
                pre1 = A[i]
                A[cnt] = A[i]
                cnt = cnt + 1
        return cnt

File: remove-duplicates-from-sorted-array-ii.java
public class Solution {
    public int removeDuplicates(int[] A) {
        Arrays.sort(A);
        int cnt = 0;
        int pre1 = 0, pre2 = 0;
        for (int i = 0; i < A.length; i++) {
            if (i < 2) {
                A[cnt++] = A[i];
                if (i == 0) pre2 = A[i];
                else pre1 = A[i];
            } else {
                if (pre1 == A[i] && pre2 == A[i])
                    continue;
                A[cnt++] = A[i];
                pre2 = pre1;
                pre1 = A[i];
            }
        }
        return cnt;
    }
}